### PR TITLE
Ammar Vehbi - JS3 Homework Week3

### PR DESCRIPTION
**Pull request by ammarvehbi**

### CODE REVIEW

File: homework/App.js [hunk 1]
```diff
@@ -15,15 +15,38 @@ class App {
...
+    });
+    select.addEventListener('change', () => this.fetchContributorsAndRender(select.value));
+    Util.createAndAppend('div', root, {
+      id: 'container',
+    });
     try {
-      const repos = await Util.fetchJSON(url);
-      this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      const response = await fetch(url);
```
Comment (yash-kapila) [inline homework/App.js]: I will also be a bit lenient here but you were actually supposed to use the `fetchJSON` method inside the Utils class to fetch repository and contributors information. 😉

File: homework/index.js [hunk 2]
```diff
@@ -30,18 +17,138 @@
...
+      const index = select.selectedIndex;
+      presentRepositories(repositoriesContainer, repositories[index]);
+      presentContributors(contributorsContainer, repositories[index].contributors_url);
+    });
+  }
+
+  async function main(url) {
+    const response = await fetch(url);
+    const data = await response.json();
+    try {
```
[the remark this answers is not in the document]
Comment (ammarvehbi) [reply homework/index.js]: Dear Yash,
Yes I have realized that but I thought the index.js file is not important. I made my final changes on other files. But I corrected the code in index.js as you suggested. Thanks @yash-kapila :pray:

File: homework/index.js [hunk 3]
```diff
@@ -30,18 +17,138 @@
...
+      });
+      li.addEventListener('click', () => {
+        window.open(contributor.html_url, '_blank');
+      });
+    });
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+  async function presentContributors(contributorsContainer, url) {
+    const response = await fetch(url);
```
[the remark this answers is not in the document]
Comment (ammarvehbi) [reply homework/index.js]: The changes are done as you suggested @yash-kapila :pray:

File: homework/App.js [hunk 4]
```diff
@@ -15,15 +15,38 @@ class App {
     // Add code here to initialize your app
     // 1. Create the fixed HTML elements of your page
     // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
-
     const root = document.getElementById('root');
-
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
-
+    // TODO: replace with your own code
```
[the remark this answers is not in the document]
Comment (ammarvehbi) [reply homework/App.js]: Done :)

File: homework/App.js (hunk 1 above)
Comment (ammarvehbi) [reply homework/App.js]: :smile:  Yes. The reason that I have used fetch is, Joost told us in the previous lecture to use fetch, so I thought that I have to use fetch here also.

**Comment by ammarvehbi** [comment]
Dear Yash, @yash-kapila 

I made changes as you suggested. Thank you very much for your valuable and constructive feedback. :pray:  :slightly_smiling_face:

**Pull request closed without merging**